### PR TITLE
Add checkbashisms package

### PR DESCRIPTION
**Pull request by lsix**
Hi

this is my first package added to Nixpkgs, so I think there is room for improvements. Comments are welcome !

Also, I was not sure where in `pkgs/top-level/all-packages.nix` was the best place to add the new package.

### CODE REVIEW

File: pkgs/development/tools/misc/checkbashisms/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  name = "checkbashisms";
+  version = "2.0.0.2";
+
+  src = fetchurl {
+    url = "http://downloads.sourceforge.net/project/checkbaskisms/${version}/${name}";
+    sha256 = "1vm0yykkg58ja9ianfpm3mgrpah109gj33b41kl0jmmm11zip9jd";
+  };
+
+  builder = builtins.toFile "builder.sh" ''
```
Comment (lethalman) [inline pkgs/development/tools/misc/checkbashisms/default.nix]: Don't use builder. You could put the mkdir, sed and chmod in `installPhase`.

**Comment by lethalman** [comment]
If your package is under tools/misc, then put it near other tools/misc :)

**Comment by lancelotsix** [comment]
Updated. Thanks for feedback.

**Comment by lethalman** [comment]
Ok thanks, thought that fixupPhase shouldn't have been needed.

**Comment by lancelotsix** [comment]
From the doc (I finally found the `nixpkgs` doc which is distinct from the `nix` doc itself) I thought this was automatically added as `perl` is an argument of the derivation.

After my tests, I found that the script was still refering the `/usr/bin/perl` binary.

I did not have a look at how the perl substitution magic happens, but is it possible that it is missed because the hedear file is `#! /usr/bin/perl` (with a ` ` between `!` and `/`) ?

**Comment by peti** [comment]
/me recommends `haskellngPackages.ShellCheck`: http://www.shellcheck.net/.

**Pull request merged**